main/src/ca/uwaterloo/flix/language/phase/Parser.scala

```diff
@@ -588,7 +588,19 @@ class Parser(val source: Source) extends org.parboiled2.Parser {
     }
 
     def Deref: Rule1[ParsedAst.Expression] = rule {
-      (SP ~ atomic("deref") ~ WS ~ Cast ~ SP ~> ParsedAst.Expression.Deref) | Cast
+      (SP ~ atomic("deref") ~ WS ~ VecSlice ~ SP ~> ParsedAst.Expression.Deref) | VecSlice
+    }
+
+    def VecSlice: Rule1[ParsedAst.Expression] = rule {
+      VecLoad ~ optional(optWS ~ atomic("V[")~ optWS ~ Literals.Int ~ optWS ~ atomic("..") ~ optWS ~ Literals.Int ~ optWS ~ "]" ~ SP ~> ParsedAst.Expression.VecSlice)
+    }
```

Comment:
This is very nice and shows that you have a grasp on how to work with the parser yourself!

I wonder if it would not be nice to permit even more syntactic sugar, i.e. all the following would be allowed:

```
e[1..2]
e[..2]
e[1..]
```

During parsing you would have to treat these separately, but after the weeder you can compile them into one ArraySlice.